### DISCUSSION TITLE
Integrated Bevy into a third party iOS app

### DISCUSSION THREAD

**Discussion by Schr3da** [bevyengine/bevy#4990] [Q&A]
Hi,

I have a quick question about how to implement bevy in an existing third party iOS app. Im aware of how to link everything properly form xcode setup perspective but I would like to know if I just can call main_rs and put the UIKit Elements on top.

Alternatively is it possible that you can share the link to the iOS code generation which might help a bit more.

kind regards
Schr3da

**Comment by bjorn3** [comment]
Winit drives the main event loop: https://github.com/rust-windowing/winit/blob/6474891f1e04d66704e3ed84ee585cd8b02ba19f/src/platform_impl/ios/event_loop.rs#L123 I don't think you can use it inside an existing app for that reason. Bevy is very modular, so it should be possible to write a replacement for bevy_winit that handles integration with an existing app though.

**Comment by Schr3da** [comment]
Thanks for the quick reply - will close the ticket now
